src/setup-ros-windows.ts

```diff
@@ -58,15 +58,18 @@ async function prepareRos2BuildEnvironment() {
 async function prepareRos2BinaryReleases() {
 	for (const rosDistro of utils.getRequiredRosDistributions()) {
 		if (rosDistro in binaryReleases) {
+			const binaryURL = binaryReleases[rosDistro];
+			const filename = binaryURL.split("/").slice(-1)[0];
+			const extension = filename.split(".").slice(1).join(".");
 			await utils.exec("wget", [
 				"--quiet",
-				binaryReleases[rosDistro],
+				binaryURL,
 				"-O",
-				`${rosDistro}.zip`,
+				`${rosDistro}.${extension}`,
 			]);
 			await utils.exec("7z", [
 				"x",
-				`${rosDistro}.zip`,
+				`${rosDistro}.${extension}`,
 				"-y",
```

Comment:
I wrote this initially because the [latest windows foxy binary release](https://github.com/ros2/ros2/releases/tag/release-foxy-20210902) was a `tar.bz2` instead of a `zip` but it looks like its not actually a `tar.bz2`... So technically this block of changes is not strictly needed for things to work, but doesnt hurt either.

more info: https://github.com/ros2/ros2/issues/1192